XmlSupportPkg/Library/XmlTreeLib/fasterxml/xml_xmlstructure.c

```diff
@@ -2317,7 +2318,8 @@ RtlXmlExtentToString (
     // Two chars required
     //
     else if (Character < 0x110000) {
-      if ((pwszWriteEnd + 2) <= pwszWriteEnd) {
+      TempPtrEnd = (UINTN)pwszWriteEnd;
+      if ((TempPtrEnd + 2 * sizeof (*pwszWriteEnd)) <= TempPtrEnd) {

```

Comment:
Arithmetic clarifcation

```suggestion
      if ((TempPtrEnd + (2 * sizeof (*pwszWriteEnd))) <= TempPtrEnd) {
```